src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -123,6 +166,83 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
         return list != null ? ImmutableList.copyOf(list) : null;
     }
 
+    /**
+     * Gets the ocean biome entries for a type of ocean.
+     * This method should be called to get the ocean biome entries for a type of ocean.
+     * This method is threadsafe.
+     *
+     * @param type The type of ocean to get the entries for.
+     * @return An ImmutableList of the ocean biome entries for the ocean type or null if there are no entries.
+     */
+    @Nullable
+    public static ImmutableList<OceanBiomeEntry> getOceanBiomes(OceanType type)
+    {
+        int idx = type.ordinal();
+        synchronized (BiomeManager.class)
+        {
+            List<OceanBiomeEntry> list = idx >= oceanBiomes.length ? null : oceanBiomes[idx];
+            return list != null ? ImmutableList.copyOf(list) : null;
+        }
+    }
+
+    /**
+     * Gets the ocean biome entries that have a biome as their main biome ('main' meaning the biome that is not its deep or mix variant).
+     * This method should be called to get the ocean biome entries that have a biome as their main biome.
+     * This method is threadsafe.
+     *
+     * @param biome The biome to check for.
+     * @param type A specific OceanType to also check for or null for any OceanType.
+     * @param excludeWarm True to exclude WARM OceanType (used internally for mixing oceans)
+     * @return An ImmutableList of ocean biome entries that meet the conditions specified in the params or null if empty
+     */
+    @Nullable
+    public static ImmutableList<OceanBiomeEntry> getOceanBiomesContainingBiome(Biome biome, @Nullable OceanType type, boolean excludeWarm)
+    {
+        OceanType[] values = type != null ? new OceanType[] {type} : OceanType.values();
+        List<OceanBiomeEntry> list = new ArrayList<>();
+        synchronized (BiomeManager.class)
+        {
+            for(OceanType types : values)
+            {
+                for(OceanBiomeEntry entry : getOceanBiomes(types))
+                {
+                    if(entry.biome == biome && (excludeWarm ? type != OceanType.WARM : true))
+                    {
+                        list.add(entry);
+                    }
+                }
+            }
+            return list.isEmpty() ? null : ImmutableList.copyOf(list);
```

Comment:
You are building this list and then just create a copy of it. Why? Use `ImmutableList.builder`.

Reply:
True, I will fix that.